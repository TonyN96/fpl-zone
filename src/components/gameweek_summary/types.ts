export interface SummaryStatType {
  label: string;
  teamCode?: number;
  playerName?: string;
  value?: string | number;
}
