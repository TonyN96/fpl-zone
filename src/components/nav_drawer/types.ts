export interface MenuItemType {
  id: string;
  label: string;
  type: "nav" | "auth";
  onClick: () => void;
}
