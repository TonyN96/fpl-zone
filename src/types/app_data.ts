import { Fixture, GameSettings, Gameweek, Phase, Player, PlayerStat, Position,Team } from "types";

export interface AppData {
  gameweeks: Gameweek[];
  gameSettings: GameSettings;
  phases: Phase[];
  teams: Team[];
  playerCount: number;
  players: Player[];
  playerStats: PlayerStat[];
  positions: Position[];
  fixtures: Fixture[];
  isCompact: boolean;
}
