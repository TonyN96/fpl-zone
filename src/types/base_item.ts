import { Player } from "./player";
import { Team } from "./team";

export type BaseItem = Player | Team;

export const isPlayer = (item: BaseItem): item is Player => {
  return "web_name" in item;
};
