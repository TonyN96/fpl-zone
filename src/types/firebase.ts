export interface FirebaseResponse {
  code: string;
  message: string;
  name: string;
}
