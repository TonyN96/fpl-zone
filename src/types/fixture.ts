export interface Fixture {
  id: number;
  code: number;
  team_h: number;
  team_h_score?: unknown;
  team_h_difficulty: number;
  team_a: number;
  team_a_score?: unknown;
  team_a_difficulty: number;
  event: number | null;
  started: boolean | null;
  finished: boolean;
  minutes: number;
  provisional_start_time: boolean;
  kickoff_time: string | null;
  stats: unknown[];
  finished_provisional?: boolean;
  pulse_id: number;
}
