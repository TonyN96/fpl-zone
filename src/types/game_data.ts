import { GameSettings } from "./game_settings";
import { Gameweek } from "./gameweek";
import { Phase } from "./phase";
import { Player } from "./player";
import { PlayerStat } from "./player_stat";
import { Position } from "./position";
import { Team } from "./team";

export interface GameData {
  events: Gameweek[];
  game_settings: GameSettings;
  phases: Phase[];
  teams: Team[];
  total_players: number;
  elements: Player[];
  element_stats: PlayerStat[];
  element_types: Position[];
}
