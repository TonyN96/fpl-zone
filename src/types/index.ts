export * from "./app_data";
export * from "./fixture";
export * from "./game_data";
export * from "./game_settings";
export * from "./gameweek";
export * from "./league";
export * from "./lineup";
export * from "./phase";
export * from "./player";
export * from "./player_past";
export * from "./player_performance";
export * from "./player_stat";
export * from "./player_summary";
export * from "./position";
export * from "./region";
export * from "./team";
export * from "./team_data";
export * from "./team_history";
export * from "./team_picks";
export * from "./transfer";
