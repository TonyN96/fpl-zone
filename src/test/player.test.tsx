import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import { mockPlayers } from "./fixture-data";
import Player from "components/player/player";
import { Player as PlayerType } from "types";
import "@testing-library/jest-dom/extend-expect";

describe("Player Tests", () => {
  let mockPlayer: PlayerType;
  let mockMultiplier: number;
  let mockIsCaptain: boolean;
  let mockIsViceCaptain: boolean;

  const mockHandlePlayerPerformanceClick = (): void => console.log("TEST");

  const createComponent = (): JSX.Element => {
    return (
      <Player
        player={mockPlayer}
        handlePlayerPerformanceClick={mockHandlePlayerPerformanceClick}
        compressed={false}
        multiplier={mockMultiplier}
        isCaptain={mockIsCaptain}
        isViceCaptain={mockIsViceCaptain}
      />
    );
  };

  beforeEach(() => {
    mockPlayer = mockPlayers[0];
    mockMultiplier = 1;
    mockIsCaptain = false;
    mockIsViceCaptain = false;
  });

  afterEach(cleanup);

  it("Snapshot test", () => {
    const { asFragment } = render(createComponent());
    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders player name as expected", () => {
    render(createComponent());

    const playerNameText = screen.getByTestId("player-name");
    expect(playerNameText).toHaveTextContent(mockPlayer.web_name.toUpperCase());
  });

  it("Renders player score as expected", () => {
    render(createComponent());

    const playerScoreText = screen.getByTestId("player-score");
    expect(playerScoreText).toHaveTextContent(mockPlayer.event_points.toString());
  });

  it("Renders player kit image as expected", () => {
    render(createComponent());

    const kitImageContainer = screen.getByTestId("kit-img-container") as HTMLImageElement;
    const url = `${process.env.PUBLIC_URL}/assets/images/kits/${mockPlayer.team_code}.png`;
    expect(kitImageContainer).toHaveStyle(`background-image: url(${url})`);
  });

  it("Multipler works as expected", () => {
    mockMultiplier = 2;

    render(createComponent());

    const playerScoreText = screen.getByTestId("player-score");
    const multipliedScore = mockPlayer.event_points * mockMultiplier;

    expect(playerScoreText).toHaveTextContent(multipliedScore.toString());
  });

  describe("Renders armband as expected", () => {
    it("Renders no armband if player is neither captain or vice", () => {
      render(createComponent());

      expect(screen.queryByTestId("armband-container")).toBeNull();
    });

    it("Renders captain armband if player captain", () => {
      mockIsCaptain = true;

      render(createComponent());

      expect(screen.getByTestId("armband-container")).toHaveTextContent("C");
    });

    it("Renders vice captain armband if player vice captain", () => {
      mockIsViceCaptain = true;

      render(createComponent());

      expect(screen.getByTestId("armband-container")).toHaveTextContent("V");
    });
  });
});
