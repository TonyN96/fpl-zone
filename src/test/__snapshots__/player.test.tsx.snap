// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
    data-testid="828"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <img
        alt="kit-img"
        data-testid="player-shirt-image"
        height="55"
        src="/assets/images/kits/3.png"
      />
    </div>
    <div
      class="MuiBox-root css-1gn4x21"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
          data-testid="player-name"
        >
          JANKIN
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
          data-testid="player-score"
        >
          13
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
